src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -457,6 +463,23 @@ private void EmitOutput(OutputSymbol outputSymbol)
             writer.WriteEndObject();
         }
 
+        public void EmitMetadata(string? templateHash, string assemblyVersion)
+        {
+            writer.WritePropertyName("metadata");
+            writer.WriteStartObject();
+            writer.WritePropertyName("_generator");
+            writer.WriteStartObject();
+            
+            this.emitter.EmitProperty("name", ThisAssembly.AssemblyName);
```

Comment:
I think we should just hard code to e.g. just "Bicep" rather than using the assembly name. Assembly name might change, and "Bicep.Core" doesn't seem intuitive to someone not familiar with the project.